### PR TITLE
docs: improve docs.sourcegraph.com homepage; clarify deployment types

### PR DESCRIPTION
**Pull request by slimsag**
@beyang I am tagging you since I'm not sure who else can approve this change. I think this is a strong improvement over our current docs homepage, and better reflects where we are at / where we are going with Delivery.

Very open to feedback from others here, but I need to move quick with this change.

## The homepage

**Screenshot: [image](https://user-images.githubusercontent.com/3173176/207742866-acf7596f-59c5-4c5c-a3c1-72c764a3c37a.png)**

* Removed Docker Compose deployment links
* Replaced Azure, DigitalOcean, and "Private Cloud" links to go to the new script installer instead (stop pushing Docker Composer.)
* Clarified that Kubernetes is for our largest enterprise customers.
* Embolden that sourcegraph.com/search can be used to try the product instantly.
* Reduce page clutter and improve legibility.
* Made "Test on local machine" deployment options a footnote, as most are confusing right now / you'd be better off using the "Script install" option on a Linux machine somewhere.

## Docker Compose

All Docker Compose pages have been updated to have "🚧 Still supported, but no longer recommended. 🚧" at the top:

image

## Test plan

Docs change only, no testing required.

### CODE REVIEW

File: doc/admin/deploy/docker-compose/aws.md [hunk 1]
```diff
@@ -1,5 +1,13 @@
 # Install Sourcegraph on Amazon Web Services (AWS)
 
+## 🚧 Still supported, but no longer recommended. 🚧
```
Comment (beyang) [inline doc/admin/deploy/docker-compose/aws.md:3]: This addition is a bit wordy and repetitive. Can we change to

> :warning: We recommend new users use [AWS AMI](path/to/new/ami/docs) or follow the [k3s-based instructions](pathto/k3s/docs), because these are easier and offer more flexibility when configuring Sourcegraph. Existing customers can reach out to CE if they wish to migrate to these deployment models.

Ditto for the other pages (I'd link directly to the corresponding $CLOUD_PROVIDER AMI docs for doc UX.

File: doc/index.md [hunk 2]
```diff
@@ -84,29 +74,6 @@ Sourcegraph is runnable in a variety of environments, from cloud to self-hosted
   </a>
 </div>
 
-### Local machine
```
Comment (beyang) [inline doc/index.md:87]: Can we keep this section and rename to "Non-production"? Primarily, I don't want to hide the single Docker container option, because I suspect folks who want to spin Sourcegraph up to try out will opt for this and it's still the best way for them to "kick the tires" and the first step to engaging with us. As a footnote, it will be too invisible to the reader.

**Comment by slimsag** [comment]
@beyang done exactly as you said, can you approve so I can merge?

**Review by abeatrix — APPROVED** [review]
